contracts/tokens/RewardEthToken.sol

```diff
@@ -162,23 +177,27 @@ contract RewardEthToken is IRewardEthToken, OwnablePausableUpgradeable, ERC20Per
     function updateTotalRewards(uint256 newTotalRewards) external override {
         require(msg.sender == balanceReporters, "RewardEthToken: access denied");
 
-        uint256 periodRewards = newTotalRewards.sub(totalRewards, "RewardEthToken: invalid new total rewards");
+        uint128 _curTotalRewards = curTotalRewards; // gas savings
+        uint256 periodRewards = newTotalRewards.sub(_curTotalRewards, "RewardEthToken: invalid new total rewards");
         if (periodRewards == 0) {
             // no new rewards
             return;
         }
 
         // calculate reward per token used for account reward calculation
         uint256 maintainerReward = periodRewards.mul(maintainerFee).div(10000);
-        uint256 newRewardPerToken = uint256(rewardPerToken).add(periodRewards.sub(maintainerReward).mul(1e18).div(stakedEthToken.totalDeposits()));
+        uint128 _curRewardPerToken = curRewardPerToken; // gas savings
+        uint128 newRewardPerToken = uint256(_curRewardPerToken).add(periodRewards.sub(maintainerReward).mul(1e18).div(stakedEthToken.totalDeposits())).toUint128();
```

Comment:
Division by `stakedEthToken.totalDeposits()` in computing `newRewardPerToken` here is problematic, especially when new deposits are added (via `Pool.addDeposit()`) after this transaction but before the beginning of the next block.

For example, suppose `periodRewards` = 1, and `stakedEthToken.totalDeposits()` = 1 at this point.  (For simplicity, assume that `_curRewardPerToken` = `maintainerReward` = 0, and ignore the precision factor `1e18`.)  Then `newRewardPerToken` = 1.

Now, suppose that new deposits are added later _within the same block_ so that `stakedEthToken.totalDeposits()` = 2 at the end of the current block.  Then, 2 rwETH tokens will be minted in the next block, while the newly generated reward was only 1.